prov/verbs/src/ep_rdm/verbs_tagged_ep_rdm.c

```diff
@@ -626,8 +627,15 @@ fi_ibv_rdm_process_recv_wc(struct fi_ibv_rdm_ep *ep, struct ibv_wc *wc)
 
 	FI_IBV_DBG_OPCODE(wc->opcode, "RECV");
 
-	if (wc->status != IBV_WC_SUCCESS ||
-	    (wc->opcode != IBV_WC_RECV_RDMA_WITH_IMM))
+	assert(wc->opcode != IBV_WC_RDMA_READ);
+
+	if (wc->status != IBV_WC_SUCCESS
+#if ENABLE_DEBUG
+	    ||
+	    (wc->opcode != IBV_WC_RECV_RDMA_WITH_IMM &&
+	     (wc->opcode != IBV_WC_RECV))
+#endif /* ENABLE_DEBUG */
+	   )
```

Comment:
Please restructure this to do something like:
```c
#if ENABLE_DEBUG
#define INVALID_OPCODE(opcode) (opcode != RDMA_WITH_IMM && opcode != RECV)
#else
#define INVALID_OPCODE(opcode) 0
#endif
```
Then modify the if statement to call INVALID_OPCODE.  This will make the code much easier to read.